src/transformers/models/bart/modeling_tf_bart.py

```diff
@@ -922,7 +940,14 @@ def call(
         past_key_values_length = shape_list(past_key_values[0][0])[2] if past_key_values is not None else 0
 
         # embed positions
-        positions = self.embed_positions(input_shape, past_key_values_length)
+        if position_ids is None:
+            if past_key_values is not None:
+                raise ValueError("Make sure to provide `decoder_position_ids` when passing `past_key_values`.")
```

Comment:
This guard was copied from Flax

Reply:
I feel a bit misleading about names: the error message is `decoder_position_ids`, but the argument is `position_ids` (for `TFBartDecoder` here).

Reply:
(I see. It means pass `decoder_position_ids` to the top level model, like `TFBartModel`)

I am OK, but thinking we can improvement the error message in the future.

Reply:
That is true 🤔 Reworded into "Make sure to provide the position ids when passing `past_key_values`.", which is always true